src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -34,16 +34,6 @@ internal sealed class BuildServerConnection
         /// <summary>
         /// Create a build request for processing on the server. 
         /// </summary>
-        /// <remarks>
-        /// Even though compilation itself does not specifically require a temporary directory to be successful
-        /// this API deliberately requires <paramref name="tempDirectory"/> to have a value. The reason for this is
-        /// that the server itself requires a temporary directory to startup and it uses the same logic as the client
-        /// to calculate it.
-        /// 
-        /// That means if the client can't calculate the temporary directory the server won't be able to either and 
-        /// hence won't be able to start. So creating <see cref="BuildRequest"/> instances isn't useful cause it's 
-        /// wasting time trying to start a server that won't be able to start.
```

Comment:
Does this need some other comment then explaining tempDirectory?

Reply:
The history of this comment is I added at the start of an earlier PR. Then some feedback came in and I realized this invariant was wrong and restructured the PR but I forgot to delete the comment. It was pointed out to me post merge and I deleted it at the start of this PR. 

So basically this never should have existed in the first place.